src/charm.py

```diff
@@ -428,20 +417,20 @@ def set_active_status(self):
     def on_install(self, event):
         """Installation routine.
 
-        Execute the Early Network stage if requested. The config of Tigera itself depends on k8s deployed
-        and should be executed in the config-changed hook.
+        Execute the Early Network stage if requested. The config of Tigera itself depends on k8s
+        deployed and should be executed in the config-changed hook.
         """
         if not self.model.config["disable_early_network"]:
             self.implement_early_network()
 
     def on_update_status(self, event):
-        """State machine of the charm.
+        """Update status.
 
-        1) Only change the status if in active
-        2) If kubeconfig and CNI relation exist
+        Unit must be in a configured state before status updates are made.
         """
-        if isinstance(self.unit.status, ActiveStatus):
-            log.info(f"on_update_status: unit not in active status: {self.unit.status}")
+        if not cast(bool, self.stored.tigera_configured):
+            log.info("on_update_status: unit has not been configured yet; skipping status update.")
+            return
```

Comment:
it almost feels like this charm should use the reconciled pattern.  But that'd be a larger refactor here than we want to make at the moment.   It seems this can get set to `False` on the leader during a config changed hook.  I like using it more than active status

Reply:
yeah, this is def a candidate for the 1.29+ charm pattern.  but as you noted, that's a larger body of work that won't make it for this maint release.

Reply:
> It seems this can get set to False on the leader during a config changed hook

correct.  if something like the license changes or a new invalid repo credential is set, config-changed will block the charm and update-status will short circuit here so that blocked status doesn't get clobbered by this function.